app/main.py

```diff
@@ -1,3 +1,55 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int):
+        self.km = km
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers"
+
+    def __add__(self, other):
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other):
+        if not isinstance(other, Distance):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other):
+        return Distance(self.km * other)
+
+    def __truediv__(self, other):
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other):
+        if not isinstance(other, Distance):
+            return True if self.km < other else False
+        return True if self.km < other.km else False
+
+    def __gt__(self, other):
+        if not isinstance(other, Distance):
+            return True if self.km > other else False
+        return True if self.km > other.km else False
+
+    def __eq__(self, other):
+        if not isinstance(other, Distance):
+            return True if self.km == other else False
+        return True if self.km == other.km else False
+
+    def __le__(self, other):
+        if not isinstance(other, Distance):
+            return True if self.km <= other else False
+        return True if self.km <= other.km else False
+
+    def __ge__(self, other):
+        if not isinstance(other, Distance):
+            return True if self.km >= other else False
```

Comment:
`other` can be only of type `float`, `int`, `Distance`. Consider this throughout your code